### ISSUE TITLE
Add `default` filter

### ISSUE THREAD

**Issue by miketaber** [dotliquid/dotliquid#179]
Maybe I'm missing something here, but I don't see anywhere in the code that the default filter is supported. I've run tests using both an AnonymousObject and a custom Drop class, but neither will recognize the something like: "{{ testvarname | default: 'defaultData' }}"

I was able to get this working by building a custom Filter as follows:
```
    public static class DefaultFilter
    {
        public static string Default(string input, string @defaultValue)
        {
            return !string.IsNullOrWhiteSpace(input) ? input : defaultValue;
        }
    }

```
It seems like this should be included in the base library but I don't see it anywhere. Am I missing something?

**Comment by laedit** [comment]
Thanks for the report.
It is missing in DotLiquid, can you open a PR with your filter? 😃

**Comment by miketaber** [comment]
Just saw this. Did someone fix this or was it closed because you were waiting on a pull request?

**Comment by haf** [comment]
Hi @miketaber – just because there was a month without a reply.

**Comment by miketaber** [comment]
I'm not terribly familiar with Git, so this took me a while to get it "working". Apologies if I do anything incorrectly here.

I've issued the pull request and added the corresponding unit tests.

I do have one of the existing unit tests which is failing on my own machine related to date formatting. I don't know why that is. I fixed it and after a push, the build server broke, but apparently it's an issue on my local system. I double checked the math for the date and the unit test should return 01 but on my system it's returning 02. I've reverted that change, but it's still broken locally. Seems to be a core issue in .NET, since after doing the math by hand, it should return 01 locally and it doesn't.
